driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -930,6 +933,214 @@ public Date deserialize(ByteBuffer bytes) {
+        private static Long parseTimeStrictly(String s) throws IllegalArgumentException
+        {
+            String nanos_s;
+
+            long hour;
+            long minute;
+            long second;
+            long a_nanos = 0;
+
+            String formatError = "Timestamp format must be hh:mm:ss[.fffffffff]";
+            String zeros = "000000000";
+
+            if (s == null)
+                throw new java.lang.IllegalArgumentException(formatError);
+            s = s.trim();
+
+            // Parse the time
+            int firstColon = s.indexOf(':');
+            int secondColon = s.indexOf(':', firstColon+1);
+
+            // Convert the time; default missing nanos
+            if (firstColon > 0 && secondColon > 0 && secondColon < s.length() - 1)
+            {
+                int period = s.indexOf('.', secondColon+1);
+                hour = Integer.parseInt(s.substring(0, firstColon));
+                if (hour < 0 || hour >= 24)
+                    throw new IllegalArgumentException("Hour out of bounds.");
+
+                minute = Integer.parseInt(s.substring(firstColon + 1, secondColon));
+                if (minute < 0 || minute >= 60)
+                    throw new IllegalArgumentException("Minute out of bounds.");
+
+                if (period > 0 && period < s.length() - 1)
+                {
+                    second = Integer.parseInt(s.substring(secondColon + 1, period));
+                    if (second < 0 || second >= 60)
+                        throw new IllegalArgumentException("Second out of bounds.");
+
+                    nanos_s = s.substring(period + 1);
+                    if (nanos_s.length() > 9)
+                        throw new IllegalArgumentException(formatError);
+                    if (!Character.isDigit(nanos_s.charAt(0)))
+                        throw new IllegalArgumentException(formatError);
+                    nanos_s = nanos_s + zeros.substring(0, 9 - nanos_s.length());
+                    a_nanos = Integer.parseInt(nanos_s);
+                }
+                else if (period > 0)
+                    throw new IllegalArgumentException(formatError);
+                else
+                {
+                    second = Integer.parseInt(s.substring(secondColon + 1));
+                    if (second < 0 || second >= 60)
+                        throw new IllegalArgumentException("Second out of bounds.");
+                }
+            }
+            else
+                throw new IllegalArgumentException(formatError);
+
+            long rawTime = 0;
+            rawTime += TimeUnit.HOURS.toNanos(hour);
+            rawTime += TimeUnit.MINUTES.toNanos(minute);
+            rawTime += TimeUnit.SECONDS.toNanos(second);
+            rawTime += a_nanos;
+            return rawTime;
+        }
+
+        private static Long parseTime(String value) throws ParseException {
+            // nano since start of day, raw
+            if (timePattern.matcher(value).matches())
```

Comment:
This case is already handled in `parse`, which is the only caller of `parseTime`. We can remove this block and the `timePattern` constant.